### PR TITLE
chore(build): validate if exporting is correct in `package.json` and `jsr.json`

### PR DESCRIPTION
**Pull request by EdamAme-x**
Related: https://github.com/honojs/hono/pull/3636

If there is a difference between the two exports, the build will fail.

### CODE REVIEW

**Comment by EdamAme-x** [comment]
It fails as follows.
image

The test succeeds when this PR is merged.
https://github.com/honojs/hono/actions/runs/11704963747/job/32598601119?pr=3638

**Comment by EdamAme-x** [comment]
Fixed some missing exports

**Comment by EdamAme-x** [comment]
@yusukebe 
Could you review it?

**Comment by yusukebe** [comment]
Hi @EdamAme-x !

This PR means you added a function to validate the configuration of exporting modules in `package.json` and `jsr.json` with `validateExports`, right?

**Comment by EdamAme-x** [comment]
That is what I mean.
If an endpoint is missing that should be exported to either, throw an error.

**Comment by yusukebe** [comment]
@EdamAme-x 

Thanks! Seems to be good. But adding a test for `validateExports` like `remove-private-fields.test.ts` is better.

**Comment by EdamAme-x** [comment]
thanks, okay.

**Comment by EdamAme-x** [comment]
@yusukebe 
ready for review

File: build/remove-private-fields.test.ts
Comment (yusukebe) [inline build/remove-private-fields.test.ts:1]: Can you remove this file's change? It is unrelated to this PR.

Comment (EdamAme-x) [reply build/remove-private-fields.test.ts:1]: okay, It looks like the format that IDE foramatter did.

**Comment by EdamAme-x** [comment]
@yusukebe 
Re-ready for review

**Comment by yusukebe** [comment]
Hi @nakasyou Can you review this?

**Comment by nakasyou** [comment]
It looks good to me. An idea, what do you think about generating `exports` in `package.json` from `jsr.json`?

File: build/build.ts [hunk 1]
```diff
@@ -7,21 +7,30 @@
...
 
 const args = arg({
   '--watch': Boolean,
 })
 
 const isWatch = args['--watch'] || false
 
+const readJsonExports = (path: string) => JSON.parse(fs.readFileSync(path, 'utf-8')).exports
+
+const [packageJsonExports, jsrJsonExports] = ['./package.json', './jsr.json'].map(readJsonExports)
```
Comment (nakasyou) [inline build/build.ts:28]: I think it also should validate `typesVersions`.

**Comment by EdamAme-x** [comment]
> It looks good to me.
Thanks.
> An idea, what do you think about generating `exports` in `package.json` from `jsr.json`?
I think there is room for consideration, let's create a new issue and discuss it there.

**Review by yusukebe — APPROVED** [review]
LGTM!

File: build/build.ts (hunk 1 above)
Comment (EdamAme-x) [reply build/build.ts:28]: thanks, will work on it.

Comment (yusukebe) [reply build/build.ts:28]: @nakasyou Ah, you are right. But perhaps validating `typeVersions` is another issue? We can check whether it's correct without `jsr.json`.

@EdamAme-x What do you think of it?

Comment (EdamAme-x) [reply build/build.ts:28]: I was able to write the code, but you are right, it may be another problem.　　
I will consider this after this PR is merged.

Comment (yusukebe) [reply build/build.ts:28]: Okay. Let's merge this first.

**Comment by yusukebe** [comment]
@EdamAme-x @nakasyou 

Thanks! Let's go with this.

**Pull request merged**